File: android/app/src/main/java/com/elenza/app/tuya/TuyaAuthModule.kt
package com.elenza.app.tuya

import com.facebook.react.bridge.*

class TuyaAuthModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "TuyaAuth"
    }

    @ReactMethod
    fun sendVerificationCode(email: String, countryCode: String, promise: Promise) {
        promise.resolve("Verification code sent successfully")
    }

    @ReactMethod
    fun registerWithEmail(email: String, password: String, code: String, countryCode: String, promise: Promise) {
        val userMap = Arguments.createMap().apply {
            putString("uid", "mock_uid")
            putString("email", email)
            putString("sid", "mock_sid")
        }
        promise.resolve(userMap)
    }

    @ReactMethod
    fun loginWithEmail(email: String, password: String, countryCode: String, promise: Promise) {
        val userMap = Arguments.createMap().apply {
            putString("uid", "mock_uid")
            putString("email", email)
            putString("sid", "mock_sid")
        }
        promise.resolve(userMap)
    }

    @ReactMethod
    fun isLoggedIn(promise: Promise) {
        val userMap = Arguments.createMap().apply {
            putString("uid", "mock_uid")
            putString("email", "dev@elenza.coffee")
            putString("sid", "mock_sid")
        }
        promise.resolve(userMap)
    }

    @ReactMethod
    fun logout(promise: Promise) {
        promise.resolve("Logged out successfully")
    }

    @ReactMethod
    fun getOrCreateHome(promise: Promise) {
        val homeMap = Arguments.createMap().apply {
            putDouble("homeId", 12345.0)
            putString("name", "ELENZA Laboratory")
        }
        promise.resolve(homeMap)
    }
}

File: ios/bridges/TuyaAuthModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TuyaAuthModule, NSObject)

RCT_EXTERN_METHOD(isLoggedIn:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(loginWithEmail:(NSString *)email
                  pass:(NSString *)pass
                  regionCode:(NSString *)regionCode
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(registerWithEmail:(NSString *)email
                  pass:(NSString *)pass
                  code:(NSString *)code
                  regionCode:(NSString *)regionCode
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(sendVerificationCode:(NSString *)email
                  regionCode:(NSString *)regionCode
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(logout:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getOrCreateHome:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end
